src/jgikbase/test/idmapping/service/mapper_service_test.py

```diff
@@ -440,8 +440,13 @@ def check_get_namespaces(returned, expected):
 
 def test_create_mapping_put():
     cli, mapper = build_app()
-    resp = cli.put('/api/v1/mapping/ans/ns', headers={'Authorization': 'source tokey'},
-                   json={'aid1': 'id1', 'id2': 'id2'})
+    # this shouldn't pass if request.get_data() isn't called before checking
+    # auth, but it does. If you're actually running a server this will
+    # cause a json parse erro with the get_data() call.
+    resp = cli.put('/api/v1/mapping/ans/ns',
+                   headers={'Authorization': 'source tokey',
+                            'content-type': 'x-www-form-urlencoded'},
```

Comment:
If you only handle application/json, but the user sets x-www-form-urlencoded, i would just immediately throw an incorrect content-type error